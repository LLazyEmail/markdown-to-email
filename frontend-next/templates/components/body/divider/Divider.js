import React from 'react';
import Table from '../../html/table';
import { styles } from './styles';

const Divider = ({ children }) => (
  <Table
    className="mcnDividerBlock"
    style={styles.dividerBlock}
  >
    <tbody className="mcnDividerBlockOuter">
      <tr>
        <td className="mcnDividerBlockInner" style={styles.dividerBlockInner}>
          <Table
            className="mcnDividerContent"
            width="100%"
            style={styles.dividerContent}
          >
            <tbody>
              <tr>
                <td style={styles.td} data-testid="dividerChildrenTest">
                  {children}
                </td>
              </tr>
            </tbody>
          </Table>
        </td>
      </tr>
    </tbody>
  </Table>
);

export default Divider;
