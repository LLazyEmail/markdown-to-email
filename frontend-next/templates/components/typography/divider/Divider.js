import React from 'react';
import { styles } from './styles';

/**
 * Todo: I think this component repeat
 */

const Divider = () => (
  <table
    border={0}
    cellPadding={0}
    cellSpacing={0}
    className="mcnDividerBlock"
    style={styles.dividerBlock}
  >
    <tbody className="mcnDividerBlockOuter">
      <tr>
        <td className="mcnDividerBlockInner" style={styles.dividerBlockInner}>
          <table
            className="mcnDividerContent"
            border={0}
            cellPadding={0}
            cellSpacing={0}
            style={styles.dividerContent}
          >
            <tbody>
              <tr>
                <td style={styles.td}>
                  <span />
                </td>
              </tr>
            </tbody>
          </table>
        </td>
      </tr>
    </tbody>
  </table>
);

export default Divider;
