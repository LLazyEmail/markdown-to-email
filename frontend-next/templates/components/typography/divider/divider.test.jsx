import React from 'react';
import { create } from 'react-test-renderer';
import ReactDOM from 'react-dom';
import Divider from './index';

describe('divider/index component', () => {
  test('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Divider />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test('renders correctly', () => {
    const tree = create(<Divider><h1>Some text</h1></Divider>);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('compoents props should contain children(h1) elements', () => {
    const component = create(<Divider><h1>Some text</h1></Divider>);
    const { root } = component;
    expect(root.props.children.type).toBe('h1');
  });
});
