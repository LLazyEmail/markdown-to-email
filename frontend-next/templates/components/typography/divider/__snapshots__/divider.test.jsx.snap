// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`divider/index component renders correctly 1`] = `
<table
  border={0}
  cellPadding={0}
  cellSpacing={0}
  className="mcnDividerBlock"
  style={
    Object {
      "WebkitTextSizeAdjust": "100%",
      "borderCollapse": "collapse",
      "minWidth": "100%",
      "msTextSizeAdjust": "100%",
      "msoTableLspace": "0pt",
      "msoTableRspace": "0pt",
      "tableLayout": "fixed !important",
    }
  }
>
  <tbody
    className="mcnDividerBlockOuter"
  >
    <tr>
      <td
        className="mcnDividerBlockInner"
        style={
          Object {
            "WebkitTextSizeAdjust": "100%",
            "minWidth": "100%",
            "msTextSizeAdjust": "100%",
            "msoLineHeightRule": "exactly",
            "padding": "12px 18px",
          }
        }
      >
        <table
          border={0}
          cellPadding={0}
          cellSpacing={0}
          className="mcnDividerContent"
          style={
            Object {
              "WebkitTextSizeAdjust": "100%",
              "borderCollapse": "collapse",
              "borderTop": "2px none #EAEAEA",
              "minWidth": "100%",
              "msTextSizeAdjust": "100%",
              "msoTableLspace": "0pt",
              "msoTableRspace": "0pt",
            }
          }
        >
          <tbody>
            <tr>
              <td
                style={
                  Object {
                    "WebkitTextSizeAdjust": "100%",
                    "msTextSizeAdjust": "100%",
                    "msoLineHeightRule": "exactly",
                  }
                }
              >
                <span />
              </td>
            </tr>
          </tbody>
        </table>
      </td>
    </tr>
  </tbody>
</table>
`;
